feat(sdk): Modify Interactive Question Layout

oisincoveney opened the pull request:
Checklist for Oisin:
- check that [feat(sdk): revamp CreateQuestion and create question behaviour](https://github.com/metabase/metabase/pull/50088) still has the same functionality after

Commit:
Remove rogue semi colons

oisincoveney commented:
This change is part of the following stack:

- #50152 ◀
    - #50509

<sub>Change managed by [git-spice](https://abhinav.github.io/git-spice/).</sub>
<!-- gs:navigation comment -->

Commit:
Fix text in `AccordionList` for SDK (#50509)

sansolovyov committed:
top-level LICENSE conflicts with other files breaking uberjar build

Commit:
Fix component with bad logic

Commit:
Merge branch 'master' into sdk/question-chill-mode

Commit:
Merge branch 'master' into sdk/question-chill-mode

Commit:
Merge branch 'master' into sdk/question-chill-mode

Commit:
Fix lint error

Commit:
More test fixes but the logs are wild

Commit:
More test fixes, hopefully

Commit:
Merge branch 'master' into sdk/question-chill-mode

Commit:
Merge branch 'master' into sdk/question-chill-mode

Commit:
Merge branch 'master' into sdk/question-chill-mode

Commit:
Ensure save modal closes when saving

Commit:
Merge branch 'master' into sdk/question-chill-mode

heypoom reviewed (commented):
I just did some testing on Storybook and Shoppy with theming applied. It looks great so far! Just a couple minor theming things to fix later. I'll review the code itself tomorrow.

https://www.loom.com/share/ccd540de7c014e5fa0493743f5376bcb?sid=2e566612-2012-4c05-9670-28d668956900

oisincoveney commented on enterprise/frontend/src/embedding-sdk/components/private/InteractiveQuestion/components/ChartTypeSelectorList/ChartTypeDropdown.tsx:
```diff
@@ -0,0 +1,95 @@
+          </Menu.Item>
+        ))}
+
+        <Menu.Label>{t`Nonsensible`}</Menu.Label>
```
Good call, I really don't like that terminology. It doesn't strike me as either "Metabasey", simple, or easily indicative of what it really means.

mazameli replied:
This should just say `Other charts`

oisincoveney replied:
done

Commit:
Merge branch 'master' into sdk/question-chill-mode

Commit:
modify readme to account for new components

Commit:
Merge branch 'master' into sdk/question-chill-mode

Commit:
Merge branch 'sdk/question-chill-mode' of https://github.com/metabase/metabase into sdk/question-chill-mode

Commit:
Merge branch 'master' into sdk/question-chill-mode

Commit:
Merge branch 'master' into sdk/question-chill-mode

Commit:
Merge branch 'master' into sdk/question-chill-mode

Commit:
Fix query dropdown inconsistencies (#50644)

Fix text in `AccordionList` for SDK (#50509)

heypoom reviewed (approved):
Functionality-wise this works great! ([see other comment on GitHub](https://github.com/metabase/metabase/pull/50152#pullrequestreview-2477894228))

PS. I noticed that the viz flashes for a brief moment (recorded in the Loom), but I'm not sure if that is on the viz side of things or some re-rendering issues in the sdk.

WiNloSt commented:
2 things I notice. I think this is popover widget dependent and it'd require a lot of effort to go fix all of this, but let's clean them up as we find.

1. This kind of date filter still doesn't go back to the main filter list popover view. In the image 1 this happens after I click on an existing filter, then after I click back I got image 2.
    - ![Screenshot 2024-12-09 at 3 50 12 PM](https://github.com/user-attachments/assets/919bbd31-2907-47a8-9757-71324576f370)
    - ![Screenshot 2024-12-09 at 3 50 15 PM](https://github.com/user-attachments/assets/3e69d755-6ee1-47f2-9ff8-b7c62b1715c7)
2. didn't we use to have some kind of reset button? I just messed my question and there's no way to reset the question back to its original state, and I'm a little sad if I have to refresh the whole page for that.
![Screenshot 2024-12-09 at 3 50 28 PM](https://github.com/user-attachments/assets/a78be28c-a3e2-4ac3-8e7a-882954f9860f)

oisincoveney commented:
> 2 things I notice. I think this is popover widget dependent and it'd require a lot of effort to go fix all of this, but let's clean them up as we find.
> 
> 1. This kind of date filter still doesn't go back to the main filter list popover view. In the image 1 this happens after I click on an existing filter, then after I click back I got image 2.
>    
>    * ![Screenshot 2024-12-09 at 3 50 12 PM](https://private-user-images.githubusercontent.com/1937582/393765538-919bbd31-2907-47a8-9757-71324576f370.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM3MzUwNjYsIm5iZiI6MTczMzczNDc2NiwicGF0aCI6Ii8xOTM3NTgyLzM5Mzc2NTUzOC05MTliYmQzMS0yOTA3LTQ3YTgtOTc1Ny03MTMyNDU3NmYzNzAucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMDlUMDg1OTI2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OWZjNzY3NTU5MGY2ZmY3MWZkOTg3NzAxNjQ4ZjRjZjRiMDQ5ZGIzNTM1ZmE3N2NkOGY5YTkxMGMxZWFlMmY3ZiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.ea2kKLSP0z2tDKVP87miMfEAApEFgExGSRbmTyKoJt0)
>    * ![Screenshot 2024-12-09 at 3 50 15 PM](https://private-user-images.githubusercontent.com/1937582/393765719-3e69d755-6ee1-47f2-9ff8-b7c62b1715c7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM3MzUwNjYsIm5iZiI6MTczMzczNDc2NiwicGF0aCI6Ii8xOTM3NTgyLzM5Mzc2NTcxOS0zZTY5ZDc1NS02ZWUxLTQ3ZjItOWZmOC1iN2M2MmIxNzE1YzcucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMDlUMDg1OTI2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NWUzYTAyZmUwYjI1NTFiMmY4NjM0ODUwNDgxZDgyOWY5MDc2ODY4YzJkMTU2YTAyNjlkZDQ3YTQ2MGUwODdmZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.1riXkf-izb9jrMJQWz0W8xQAzPvgRVwmQTJk-WdeZFs)

Yep, I should have mentioned in the last PR. This behavior is way harder to fix than it looks, since the date picker is really intertwined within a deep part of the code and changing the navigation of that will take more time than I think is useful right now. I can add it to the task list of things. 

What is the expected behavior again? If I click the back button, it does straight back to the badge list, right?


> 2. didn't we use to have some kind of reset button? I just messed my question and there's no way to reset the question back to its original state, and I'm a little sad if I have to refresh the whole page for that.
>    ![Screenshot 2024-12-09 at 3 50 28 PM](https://private-user-images.githubusercontent.com/1937582/393765976-a78be28c-a3e2-4ac3-8e7a-882954f9860f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM3MzUwNjYsIm5iZiI6MTczMzczNDc2NiwicGF0aCI6Ii8xOTM3NTgyLzM5Mzc2NTk3Ni1hNzhiZTI4Yy1hM2UyLTRhYzMtOGU3YS04ODI5NTRmOTg2MGYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMDlUMDg1OTI2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NjI2YzQwNzQ1ZDUwMDMwNzQxZjZlYzgxNjQwNWYwMzYyNmI0MTAzOTFkZTg3ZGI2OWQ1MWYzYzNlNDRmMjAyOCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.FkBX6vMGs62_Jiobh-UmFIaK53kAB-MTCTypUsWZr7c)

the reset should be there - it should pop up above the title (similar to the core app) as text that says "Back to [ORIGINAL QUESTION NAME]".  I'm using the same logic as the reset button so that's super weird.

Commit:
Fix bucket button and popover text style issues (#50522)

fix(sdk): Redesign Interactive Question Notebook button (#50228)
Fix text in `AccordionList` for SDK (#50509)

Commit:
Merge branch 'sdk/question-chill-mode' of https://github.com/metabase/metabase into sdk/question-chill-mode

Commit:
Merge branch 'master' into sdk/question-chill-mode

Commit:
change 'Nonsensical' to 'Other charts'

Commit:
Ensure there's a reset button when there's a custom string title, and modify story to match

Commit:
Merge branch 'master' into sdk/question-chill-mode

Commit:
Fix type error

Commit:
Merge branch 'sdk/question-chill-mode' of https://github.com/metabase/metabase into sdk/question-chill-mode

Commit:
Merge branch 'master' into sdk/question-chill-mode

oisincoveney commented:
@metabase-bot backport release-x.51.x
